core/src/operations/mod.rs

```diff
@@ -49,6 +49,10 @@ pub enum Operation {
     /// Marks the beginning of a function call.
     Call,
 
+    /// Marks the beginning of a dynamic function call, where the target function is specified by
+    /// the top of the stack.
+    Dyn,
```

Comment:
nit: I probably wouldn't mention function calls here as we usually associate calls with context switches (i.e., via `call` or `syscall` blocks).